### ISSUE TITLE
Integrate with OpenAI function call API

### ISSUE THREAD

**Issue by HelgeSverre** [HelgeSverre/mindwave#11]
https://platform.openai.com/docs/guides/gpt/function-calling

We can base an entire agent type of of this API alone, might make more sense to simply focus the library around OpenAI and ignore other providers (for now), until we have a good working version, most people will use OpenAI anyways.

Overcomplicating it from the start is will slow the library down, laser focus on solving real problems (Agents and Deep integration with eloquent/laravel), we can go "wide" later.

TL;DR: 
- Focus mostly on OpenAI and the new function calling API.
- Make it easy to hook it into your laravel app (eloquent models, commands, jobs etc).

**Comment by HelgeSverre** [comment]
Started on this on the new-version branch, it will also support passing a closure, job or whatever and it will automatically generate a json schema to describe the parameters, then make some neat api for executing the function.

I can see use cases like being able to define 'Actions' that can be called either sync, or queued, etc, sort of like filament actions, but triggered by the ai response.
